ksql-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadata.java

```diff
@@ -134,6 +137,10 @@ public PhysicalSchema getPhysicalSchema() {
       final QueryId queryId,
       final QueryContext.Stacker contextStacker
   ) {
-    return materializationProvider.map(builder -> builder.build(queryId, contextStacker));
+    if (!materialization.isPresent()) {
+      materialization = materializationProvider.map(builder ->
```

Comment:
it looks like we're caching a value that may not have the same `queyId` that the caller is making (i.e. the first time I call it with `queryId=123` and second time I call it with `queryId=124`, it will use `123` for both of them) - digging into the code, it looks like it might cause our logging pipeline to be a little messed up. We should look at this with a critical eye and see what we can do